### PR TITLE
PW-4775 touchscreen error IE/UWP fix

### PR DESCRIPTION
**Pull request by thomhs22**
Fixed the 'Touchscreen is undefined' error on IE/UWP when using mouse and touchscreen laptop

### CODE REVIEW

**Review by jmsn — COMMENTED** [review]
Could you please elaborate what this `Touchscreen is undefined` error is caused by? I don't quite get it. @thomhs22

**Comment by thomhs22** [comment]
> Could you please elaborate what this `Touchscreen is undefined` error is caused by? I don't quite get it. @thomhs22

hello @jmsn , sure, as specified in ticket: https://bitmovin.atlassian.net/browse/PW-4775, this issue is seen on IE/UWP apps when trying to scrub the seek bar/volume bar, using the mouse, only when it is a Windows Touchscreen Device.
The `TouchEvent` type is not supported by IE (https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent/TouchEvent),  so the following instruction fails:

> let isTouchEvent = BrowserUtils.isTouchSupported && e instanceof TouchEvent;

because `TouchEvent `is undefined.
My fix is adding a validation by checking the `constructor name` before doing the `TypeAssertion`

**Comment by stonko1994** [comment]
@thomhs22 please don't use internal issue-links in this repo as it's public and open source

**Review by Cagince — APPROVED** [review]

File: src/ts/components/seekbar.ts [hunk 1]
```diff
@@ -652,7 +652,7 @@ export class SeekBar extends Component<SeekBarConfig> {
     // and mouseup handlers to the whole document. A seek is triggered when the user lifts the mouse key.
     // A seek mouse gesture is thus basically a click with a long time frame between down and up events.
     seekBar.on('touchstart mousedown', (e: MouseEvent | TouchEvent) => {
-      let isTouchEvent = BrowserUtils.isTouchSupported && e instanceof TouchEvent;
+      let isTouchEvent = BrowserUtils.isTouchSupported && e.constructor.name === 'Touchscreen' && e instanceof TouchEvent;
```
Comment (jmsn) [inline src/ts/components/seekbar.ts:655]: I'm afraid, I think this check for the constructor is not very predictable/safe and there's an easier way.

I'd propose to check for `window.TouchEvent` instead. And I'd put that in a helper function that we'd then use in both places 🙂

```ts
function isTouchEvent(e: UIEvent): e is TouchEvent {
  return window.TouchEvent && e instanceof TouchEvent;
};
```

**Review by jmsn — CHANGES_REQUESTED** [review]

File: src/ts/components/seekbar.ts (hunk 1 above)
Comment (thomhs22) [reply src/ts/components/seekbar.ts:655]: hi @jmsn unfortunately  the `window.TouchEvent` is also `undefined` on IE, I tried this as well.

Comment (jmsn) [reply src/ts/components/seekbar.ts:655]: yes and that's exactly why checking `window.TouchEvent` before `e instanceof TouchEvent` will prevent us from running into the error 🙂

Comment (thomhs22) [reply src/ts/components/seekbar.ts]: thanks for the suggestion @jmsn I just made the changes.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ### Fixed
 - Controls' focus highlighting is shown in case of non-keyboard interaction on some browsers/platforms
+- TouchEvent error in seekbar controls when using the mouse on a touchscreen Windows device and IE/UWP
```
Comment (jmsn) [inline CHANGELOG.md:11]:
```suggestion
- `TouchEvent` error in seekbar controls when using the mouse on a touchscreen Windows device and IE/UWP
```

**Review by jmsn — COMMENTED** [review]
Nice, one last nitpick and we're good I think

**Review by jmsn — APPROVED** [review]

**Pull request merged**